nomad/state/state_store.go

```diff
@@ -1894,6 +1894,80 @@ func (s *StateStore) DeleteJobTxn(index uint64, namespace, jobID string, txn Txn
 		return err
 	}
 
+	// Delete job deployments
+	deployments, err := s.DeploymentsByJobID(nil, namespace, job.ID, true)
+	if err != nil {
+		return fmt.Errorf("deployment lookup for job %s failed: %v", job.ID, err)
+	}
+
+	for _, deployment := range deployments {
+		// Lookup the deployment
+		existing, err := txn.First("deployment", "id", deployment.ID)
+		if err != nil {
+			return fmt.Errorf("deployment lookup failed: %v", err)
+		}
+		if existing == nil {
+			return fmt.Errorf("deployment not found")
+		}
```

Comment:
```suggestion
		if existing == nil {
			continue
		}
```

Since we're trying to delete deployments, if the deployment doesn't exist then our job is already done so I think we can just continue deleting the others instead.

Reply:
👍 Fixed in [d6849cc](https://github.com/hashicorp/nomad/pull/19609/commits/d6849cc849e8cbd4e5a214435a9fd98fe560b033)